cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-s2i-deployments.adoc

```diff
@@ -0,0 +1,218 @@
++
+[source,terminal]
+----
+$ oc new-app https://github.com/<UserName>/ostoy \
+    --env=MICROSERVICE_NAME=OSTOY_MICROSERVICE
+----
++
+.Example output
++
+[source,terminal]
+----
+Creating resources ...
+   imagestream "ostoy" created
+   buildconfig "ostoy" created
+   deploymentconfig "ostoy" created
+   service "ostoy" created
+Success
+   Build scheduled, use 'oc logs -f bc/ostoy' to track its progress.
+   Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
+     'oc expose svc/ostoy'
+   Run 'oc status' to view your app.
+----
+
+. Update the Deployment
++
+Update the deployment to use a "Recreate" deployment strategy (as opposed to the default of `RollingUpdate`) for consistent deployments with persistent volumes. Reasoning here is that the PV is backed by EBS and as such only supports the `RWO` method. If the deployment is updated without all existing pods being killed it might not be able to schedule a new pod and create a PVC for the PV as it's still bound to the existing pod. If you will be using EFS you do not have to change this.
++
+[source,terminal]
+----
+$ oc patch deployment ostoy --type=json -p \
+    '[{"op": "replace", "path": "/spec/strategy/type", "value": "Recreate"}, {"op": "remove", "path": "/spec/strategy/rollingUpdate"}]'
+----
+
+. Set a Liveness probe
++
+Create a Liveness Probe on the Deployment to ensure the pod is restarted if something isn't healthy within the application. Enter the following into the CLI:
++
+[source,terminal]
+----
+$ oc set probe deployment ostoy --liveness --get-url=http://:8080/health
+----
+
+. Attach Secret, ConfigMap, and PersistentVolume to Deployment
++
+Run the following commands attach your secret, ConfigMap, and PersistentVolume:
++
+.. Attach Secret
++
+[source,terminal]
+----
+$ oc set volume deployment ostoy --add \
+    --secret-name=ostoy-secret \
+    --mount-path=/var/secret
+----
++
+.. Attach ConfigMap
++
+[source,terminal]
+----
+$ oc set volume deployment ostoy --add \
+    --configmap-name=ostoy-config \
+    -m /var/config
+----
+
+.. Create and attach PersistentVolume
++
+[source,terminal]
+----
+$ oc set volume deployment ostoy --add \
+    --type=pvc \
+    --claim-size=1G \
+    -m /var/demo_files
+----
+
+. Expose the UI application as an OpenShift Route
++
+Run the following command to deploy this as an HTTPS application that uses the included TLS wildcard certificates:
++
+[source,terminal]
+----
+$ oc create route edge --service=ostoy --insecure-policy=Redirect
+----
+
+. Browse to your application with the following methods:
+** Running the following command opens a web browser with your OSToy application:
++
+[source,terminal]
+----
+$ python -m webbrowser "$(oc get route ostoy -o template --template='https://{{.spec.host}}')"
```

Comment:
Do we need to add the for Mac users, they need to put a 3 after python? and that the window will automatically open?

Reply:
You shouldn't need to tell users about `python3` since they'll likely have it set correctly on their system or use `python 3` already. We already note on line 206 that:
`** Running the following command opens a web browser with your OSToy application:`